src/openvic-simulation/utility/Getters.hpp

```diff
@@ -63,4 +63,13 @@ public: \
 \
 private:
 
+#define PROPERTY_RW(NAME) \
+	PROPERTY(NAME) \
+public: \
+	void set_##NAME(decltype(NAME) new_##NAME) { \
+		NAME = new_##NAME; \
+	} \
+\
+private:
```

Comment:
Good job with how you've used this macro, only with things which we're happy to pass by value (so primitive types and basic primitive type wrappers like `Date` and `fixed_point_t`). As for special logic to choose the argument type and control how it's assigned, let's see what Spartan can come up with (but that can be a future PR).